### ISSUE TITLE
Starter

### ISSUE THREAD

**Issue by bmoneke** [pymeasure/experiment_control_protocol#11]
Tango has a nice feature, a **starter**.

That software can start/stop services. Therefore you do not need to start all the _actors_ manually, but can have a single (or several) starter instances, which run tasks/services  (for example other _actors_) in individual threads or processes. The starter can be an _actor_ itself and therefore controlled.

Multithreading is great, as an Exception in a child thread does not stop the main thread.

The starter could also regularly check (at least for specified tasks), whether they run and restart them if necessary (similar to operating system services).

This is an optional (but handy) feature.

**Comment by bilderbuchi** [comment]
Hm, I kinda envisioned the Director to start all the other components. However, I can see how that could be problematic.

**Comment by bklebel** [comment]
I think such a starter would be good, as an optional feature. I do not think we want to restrict adopters to use this starter, they may put Actors into daemon-like OS-specific things as they like, but they get the option to do it with this, in an OS-agnostic way. 
If they so want, they could have the starter bring up a Director, but if a GUI was attached to such a Director, I personally would avoid that.

**Comment by bmoneke** [comment]
Yes, I see the starter as a convenience tool to reduce the number of running programs (as they are encapsulated in the starter) and to be able to start "programs" via commands. So you can start all the device drivers you need for a specific experiment easily.
